### PR TITLE
Moving configline classes to individual class files

### PR DESCRIPTION
**Pull request by azito122**
This is the absolute minimum of changes necessary for this reorganization. I have split every individual step into a separate commit to make for easier diffing. And of course we can rebase before merge if preferred.

To summarize, I split the configline classes into files in `/classes`, and put them each into the block namespace (`block_filtered_course_list`). Then, I changed a bunch of class names and references to work with the new system.

### CODE REVIEW

File: classes/generic_configline.php [hunk 1]
```diff
@@ -67,15 +67,15 @@ public function get_rubrics() {
...
+                $this->rubrics[] = new \block_filtered_course_list_rubric($label, $list, $this->config, $this->line['expanded']);
             } else {
                 // Just print course names of single category.
                 $category = array_shift($categories);
                 $courses = get_courses($category->id);
                 if ($courses) {
                     $label = $this->line['courselistheading'];
                     $list = $courses;
-                    $this->rubrics[] = new block_filtered_course_list_rubric($label, $list, $this->config, $this->line['expanded']);
+                    $this->rubrics[] = new \block_filtered_course_list_rubric($label, $list, $this->config, $this->line['expanded']);
```
Comment (kwiliarty) [inline classes/generic_configline.php]: Line 78 ends up being one character too long for the code checker. I know this is not the only warning that needs to be cleaned up, but this one is directly related to the current change, so please shorten or split the line.

**Comment by azito122** [comment]
1. Added docblocks for files
2. Moved `$this->line['expanded']` to variable to shorten line 78
3. Replaced all `configline` with `filter`, so now we've got `generic_filter`, `shortname_filter`, etc.

**Comment by kwiliarty** [comment]
Perfect. Thank you. Go ahead and rebase to a single commit. Once the tests pass, I'll integrate it.

**Comment by kwiliarty** [comment]
This code is now integrated with the master branch

**Pull request closed without merging**